agenta-web/src/components/Scripts/GlobalScripts.tsx

```diff
@@ -1,18 +1,21 @@
-import {useState, useEffect, useCallback, ComponentType} from "react"
+import {useState, useEffect, useCallback, ComponentType, useRef} from "react"
 import Head from "next/head"
 import {isDemo} from "@/lib/helpers/utils"
 import {dynamicComponent} from "@/lib/helpers/dynamic"
 
 const GlobalScripts = () => {
     const [CloudScripts, setCloudScripts] = useState<ComponentType | null>(null)
+    const isLoading = useRef(false)
 
     const initializeScripts = useCallback(() => {
         const Scripts = dynamicComponent("Scripts/assets/CloudScripts")
         setCloudScripts(() => Scripts)
     }, [])
 
     useEffect(() => {
-        if (isDemo()) {
+        if (!isLoading.current && isDemo()) {
```

Comment:
or this could simply be just checking if Scripts are there. if there, not execute. maybe also

```
setCloudScripts(() => Scripts) -> setCloudScripts((prev) => prev || Scripts)
```

Reply:
fixed